streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -83,6 +83,14 @@ protected JoinWindows(final JoinWindows joinWindows) {
         afterMs = joinWindows.afterMs;
         graceMs = joinWindows.graceMs;
         enableSpuriousResultFix = joinWindows.enableSpuriousResultFix;
+
+        if (beforeMs + afterMs < 0) {
+            throw new IllegalArgumentException("Window interval (ie, beforeMs+afterMs) must not be negative.");
```

Comment:
> Could we make this Constructor to call another overloaded constructor, to avoid duplicated codes? i.e.
> 
> ```java
> protected JoinWindows(final JoinWindows joinWindows) {
>     this(joinWindows.beforeMs, joinWindows.afterMs, joinWindows.graceMs, joinWindows.enableSpuriousResultFix);
> }
> ```

This is done.